rest-client-base/src/main/java/com/wultra/core/rest/client/base/DefaultRestClient.java

```diff
@@ -0,0 +1,700 @@
+                    .headers(h -> {
+                        if (headers != null) {
+                            h.addAll(headers);
+                        }
+                    })
+                    .contentType(config.getContentType())
+                    .accept(config.getAcceptType());
+            response = buildResponseMono(spec, request).block();
+            validateResponse(response, responseType);
+            return response.toEntity(responseType).block();
+        } catch (RestClientException ex) {
+            // Rethrow validation errors
+            throw ex;
+        } catch (Exception ex) {
+            throw new RestClientException("HTTP POST request failed", ex);
+        }
+    }
+
+    @Override
+    public void postNonBlocking(String path, Object request, Consumer<ClientResponse> onSuccess, Consumer<Throwable> onError) throws RestClientException {
+        postNonBlocking(path, request, null, onSuccess, onError);
+    }
+
+    @Override
+    public void postNonBlocking(String path, Object request, MultiValueMap<String, String> headers, Consumer<ClientResponse> onSuccess, Consumer<Throwable> onError) throws RestClientException {
+        try {
+            WebClient.RequestBodySpec spec = buildUri(webClient.post(), path)
+                    .headers(h -> {
+                        if (headers != null) {
+                            h.addAll(headers);
+                        }
+                    })
+                    .contentType(config.getContentType())
+                    .accept(config.getAcceptType());
+            Mono<ClientResponse> responseMono = buildResponseMono(spec, request);
+            responseMono.subscribe(onSuccess, onError);
+        } catch (Exception ex) {
+            throw new RestClientException("HTTP POST request failed", ex);
+        }
+    }
+
+    @Override
+    public <T> ObjectResponse<T> postObject(String path, ObjectRequest<?> objectRequest, Class<T> responseType) throws RestClientException {
+        return postObject(path, objectRequest, null, responseType);
+    }
+
+    @Override
+    public <T> ObjectResponse<T> postObject(String path, ObjectRequest<?> objectRequest, MultiValueMap<String, String> headers, Class<T> responseType) throws RestClientException {
+        ParameterizedTypeReference<ObjectResponse<T>> typeReference = getTypeReference(responseType);
+        ResponseEntity<ObjectResponse<T>> responseEntity = post(path, objectRequest, headers, typeReference);
+        return responseEntity.getBody();
+    }
+
+    @Override
+    public <T> ResponseEntity<T> put(String path, Object request, ParameterizedTypeReference<T> responseType) throws RestClientException {
+        return put(path, request, null, responseType);
+    }
+
+    @Override
+    public <T> ResponseEntity<T> put(String path, Object request, MultiValueMap<String, String> headers, ParameterizedTypeReference<T> responseType) throws RestClientException {
+        ClientResponse response;
```

Comment:
Can we move the declaration to the assignment?

Reply:
Fixed.